### PR TITLE
New rule: Use matching super parameter names

### PR DESCRIPTION
**Pull request by srawlins**
Implements the proposal at https://github.com/dart-lang/linter/issues/4259

### CODE REVIEW

**Review by bwilkerson — APPROVED** [review]

**Comment by pq** [comment]
I'd like to get a little more conversation on the proposal (https://github.com/dart-lang/linter/issues/4259) before landing.

**Comment by srawlins** [comment]
Sounds good. Anything specific?

**Comment by pq** [comment]
As per https://github.com/dart-lang/linter/blob/main/doc/lint-lifecycle.md we generally try and get a bunch of input before "accepting" a proposed lint. I'll go ahead and cc some folks who are reliably opinionated.

File: lib/src/rules/matching_super_parameters.dart [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+}
+
+class _Visitor extends SimpleAstVisitor<void> {
+  final LintRule rule;
+
+  const _Visitor(this.rule);
+
+  @override
+  void visitConstructorDeclaration(ConstructorDeclaration node) {
+    var positionalSuperParameters = node.parameters.parameters
```
Comment (pq) [inline lib/src/rules/matching_super_parameters.dart:95]: I'd be inclined to do this less idiomatically in preference of just one iteration and without all the temporary objects. (I did a series of changes that dumbed down code like this and made it measurably faster.)

Comment (pq) [reply lib/src/rules/matching_super_parameters.dart:95]: I mean, parameter lists are short but I guess I think every bit counts. 🤔

Comment (srawlins) [reply lib/src/rules/matching_super_parameters.dart]: Done

**Pull request merged**